README.md

```diff
@@ -1,32 +1,574 @@
+have much more problems within our code, meaning that the same code could have a
+lot of different outcomes, right? Yeah, ES6 assynchrony introduces exactly that
+:troll: but don't worry, i will explain it in a minute.
+
+### imagine instagram feed - with infinite scrolling
+
+if we had to implement that:
+
+* add something like an _onscroll_ event
+* after some scroll, trigger an _ajax request_ to fetch more posts
+* when we get the response, append it to the end of the feed
+
+there are two parts of this tasks, i will call it _processes_ just for the sake
+of the task, not that they're executing in parallel
+
+process 1
+
+```
+onscroll, request 1
+onscroll, request 2
+onscroll, request 3
+onscroll, request 4
+onscroll, request 5
+onscroll, request 6
+onscroll, request 7
+```
+
+process 2
+
+```
+response 1
+response 2
+response 3
+response 4
+response 5
+response 6
+response 7
+```
+
+how it happens on the _event loop_
+
+```
+onscroll, request 1   <--- Process 1 starts
+onscroll, request 2
+response 1            <--- Process 2 starts
+onscroll, request 3
+response 2
+response 3
+onscroll, request 4
+onscroll, request 5
+onscroll, request 6
+response 4
+onscroll, request 7   <--- Process 1 finishes
+response 6
+response 5
+response 7            <--- Process 2 finishes
+```
+
+### what if we wanna process the data comming from the request
+
+* image you wanna go through all this posts and add `1 day` to their timestamp
+
+```
+...
+
+posts = posts.map(post => {
+    post.timestamp = moment(post.timestamp).add(1, 'day')
+})
+```
+
+* it would be fine, if it was like a few thousands of posts. Now if it was
+  1million? now that you know about the event loop, let's try to execute code
+  that blocks it. When the code you are executing is expensive/takes too long,
+  and you execute it syncrohonously, the event loop has no _time_ to get to the
+  next item, this means it is **only executing your task**
+
+```
+while (true) {
+    console.log('looooop')
+}
+```
+
+### better way to do this?
+
+somebody who has used node before would probably come with this approach
+
+```
+let processedPosts = []
+
+function process(posts) => {
+  const chunk = posts.splice(0, 1000)
+
+  processedPosts = processedPosts.concat(
+    chunk.map(post => post.timestamp = moment(posts.timestamp).add(1, 'day'))
+  )
+
+  if (posts.length) {
+    setTimeout(process(posts), 0)
+  }
+}
+```
+
+the _setTimeout_ 0 magic. As you may have understood, using `setTimeout` with 0
+as a timeout means we're executing our callback after whatever it is on the
+event loop (we're appending an item to the end). Giving this way a change for
+the browser to execute whatever it needs.
+
+## es6 introduces the job queue
+
+* what is the job queue? it's a queue hanging of at the end of every `tick` of
+  the loop
+
+looking back at our _dummy event loop implementation_
+
+```
+// `eventLoop` is an array that acts as a queue (first-in, first-out)
+var eventLoop = [ ];
```

Comment:
Did you deliberately choose pre-es6 code for the event loop examples?

Reply:
No, I have to convert it, good catch